clientconn.go

```diff
@@ -402,8 +402,10 @@ func (cc *ClientConn) newAddrConn(addr Address, skipWait bool) error {
 	// skipWait may overwrite the decision in ac.dopts.block.
 	if ac.dopts.block && !skipWait {
 		if err := ac.resetTransport(false); err != nil {
-			ac.tearDown(err)
-			return err
+			if err != errConnClosing {
+				ac.tearDown(err)
+			}
+			return fmt.Errorf("failed to create transport: %v", err)
```

Comment:
Done.
`Dial()` will return the error received from `TransportCredentials.ClientHandshake()`.